App/lib/hooks/usePersistStore.js

```diff
@@ -0,0 +1,50 @@
+import { useEffect } from 'react';
+import AsyncStorage from '@react-native-community/async-storage';
+import { useDispatch, useSelector } from 'react-redux';
+
+import useAppState from './useAppState';
+import useIsMounted from './useIsMounted';
+
+export default function usePersistStore() {
+  const dispatch = useDispatch();
+  const isMounted = useIsMounted();
+  const { appState } = useAppState();
+  const state = useSelector((state) => state);
+
+  useEffect(() => {
+    retrieveStore();
+
+    return () => {
+      saveStore();
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!isMounted) return;
+    if ('active' === appState) return;
+    
+    // 'background' || 'inactive
+    saveStore();
+  }, [appState]);
+  
+  function saveStore() {
+    debugger
```

Comment:
Are these `debugger` statements supposed to be here still?

Reply:
Oops, I'll remove those, thanks

Reply:
thats fixed